src/blockencodings.h

```diff
@@ -11,17 +11,60 @@
 class CTxMemPool;
 
 // Dumb helper to handle CTransaction compression at serialize-time
-struct TransactionCompressor {
-private:
-    CTransactionRef& tx;
-public:
-    explicit TransactionCompressor(CTransactionRef& txIn) : tx(txIn) {}
+struct TransactionCompression
+{
+    template<typename Stream> void Ser(Stream& s, const CTransactionRef& tx) { s << tx; }
+    template<typename Stream> void Unser(Stream& s, CTransactionRef& tx) { s >> tx; }
+};
+
+struct Uint48Formatter
+{
+    template <typename Stream, typename I> void Ser(Stream& s, I v)
+    {
+        static_assert(std::is_unsigned<I>::value, "Uint48Formatter needs an unsigned integer");
+        static_assert(sizeof(I) >= 6, "Uint48Formatter needs a 48+ bit type");
+
+        uint32_t lsb = v & 0xffffffff;
+        uint16_t msb = (v >> 32) & 0xffff;
+        s << lsb << msb;
+    }
 
-    ADD_SERIALIZE_METHODS;
+    template <typename Stream, typename I> void Unser(Stream& s, I& v)
+    {
+        static_assert(std::is_unsigned<I>::value, "Uint48Formatter needs an unsigned integer");
+        static_assert(sizeof(I) >= 6, "Uint48Formatter needs a 48+ bit type");
 
-    template <typename Stream, typename Operation>
-    inline void SerializationOp(Stream& s, Operation ser_action) {
-        READWRITE(tx); //TODO: Compress tx encoding
+        uint32_t lsb;
+        uint16_t msb;
+        s >> lsb >> msb;
+        v = (uint64_t(msb) << 32) | uint64_t(lsb);
+    }
+};
+
+struct DifferentialSerTransform
+{
+    uint16_t m_subtract = 0;
+
+    uint16_t operator()(uint16_t val)
```

Comment:
That doesn't really work, because `VectorUsing` needs the input and output of the transform to be identical (or at least compatible)...  suspect there is a way to make it infer what the input type to the transform on deserialization is, but right now, it creates a `V::value_type` to deserialize into.

Reply:
No longer relevant in new code.